src/libs/ValidationUtils.js

```diff
@@ -274,6 +274,15 @@ function isValidPassword(password) {
     return password.match(CONST.PASSWORD_COMPLEXITY_REGEX_STRING);
 }
 
+/**
+ * @param {String} code
+ * @returns {Boolean}
+ */
+function isValidTwoFactorCode(code) {
+    const result = code.match(CONST.REGEX.BASE64);
```

Comment:
why are we checking base64 here?

Reply:
Matches our WAF rule [here](https://github.com/Expensify/Web-Expensify/blob/0652e9fe38349fb17d621e3449de38866c51beb6/_inputrules.php#L965)

Reply:
Hmm that was added before we even had 2FA. It's a bit weird that we would call something valid that is not valid. 
If our 2FA checks doesn't accept letters let's not allow them here since that's the whole point of this function
cc @bondydaa

Reply:
That was added for OldDot 2FA. We use the same logic in NewDot. Our 2FA recovery codes are alphanumeric so those are valid from an API standpoint.

Reply:
I think we're talking about two different things. 
I'm saying the client-side validation for this should be whatever this input accepts. So if this input only expects numbers it should only "validate" numbers regardless of what the API validates. 
If this is also the same input we use for recover codes then this is fine.

Reply:
> If this is also the same input we use for recover codes then this is fine.

Yeah we use this two factor field for entering both normal two factor codes and recovery codes. So yes this input does expect both letters and numbers. We're just adding validation to handle special characters input

Reply:
Yeah this was the thread: https://expensify.slack.com/archives/C03U7DCU4/p1672348143934859
So I think you should also change the input to be text instead of number pad if you're going to validate recovery codes as well. 
But from that thread it looks like that will be handled somewhere separately so I think you should just validate the numbers for now.

Reply:
Tbh I think the only thing preventing proper support of recovery codes is just the character limit so it feels wrong to just add more artificial limits, but I guess that can be decided at another time.